C/permutation_of_string.c

```diff
@@ -0,0 +1,35 @@
+#include <stdio.h> 
+#include <string.h> 
+
+void swap(char *x, char *y) 
+{ 
+	char temp; 
+	temp = *x; 
+	*x = *y; 
+	*y = temp; 
+} 
+
+void permute(char *a, int l, int r) 
+{ 
+int i; 
+if (l == r) 
+	printf("%s\n", a); 
+else
+{ 
+	for (i = l; i <= r; i++) 
+	{ 
+		swap((a+l), (a+i)); 
+		permute(a, l+1, r); 
+		swap((a+l), (a+i)); 
+	} 
+} 
+} 
```

Comment:
https://www.geeksforgeeks.org/write-a-c-program-to-print-all-permutations-of-a-given-string/